### PR TITLE
Add UniDepthV2

### PR DESCRIPTION
**Pull request by lpiccinelli-eth**
This PR includes UniDepthV2 mode and weights (huggingface) support and the support to export V2 in `.onnx` format.

Docs are modified accordingly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -140,18 +141,38 @@ data = {"image": rgb, "K": intrinsics}
...
-Please visit [Hugging Face](https://huggingface.co/lpiccinelli) to access the repo models with weights. You can load V1 of UniDepth as:
+Please visit [Hugging Face](https://huggingface.co/lpiccinelli) to access the repo models with weights. You can load UniDepth as:
 
 ```python
-from unidepth.models import UniDepthV1
+from unidepth.models import UniDepthV1, UniDepthV2
 
 model = UniDepthV1.from_pretrained(f"lpiccinelli/unidepth-v1-{backbone}")
+# if you want to load V2 from HuggingFace:
+# model = UniDepthV2.from_pretrained(f"lpiccinelli/unidepth-v2-{backbone}")
```
Comment (RoyYang0714) [inline README.md:151]: I guess we don't need comments here, right?

Just model_v1 = ... and model_v2 = ...

File: assets/docs/V2_README.md
Comment (RoyYang0714) [inline assets/docs/V2_README.md:1]: Should we put a table of contents in the first paragraph?

File: scripts/demo.py [hunk 2]
```diff
@@ -38,6 +38,7 @@ def demo(model):
 if __name__ == "__main__":
     print("Torch version:", torch.__version__)
     model = UniDepthV1.from_pretrained("lpiccinelli/unidepth-v1-vitl14")
+    # model = UniDepthV2.from_pretrained("lpiccinelli/unidepth-v2-vitl14")
```
Comment (RoyYang0714) [inline scripts/demo.py:41]: Which one should we use for the demo by default? We could always use the latest version, right?

Comment (lpiccinelli-eth) [reply scripts/demo.py:41]: This is just to make people aware of it

File: assets/docs/V2_README.md
Comment (lpiccinelli-eth) [reply assets/docs/V2_README.md:1]: This is just a detailed explanation of the main page contents

**Pull request merged**